init.el

```diff
@@ -13,6 +13,54 @@
   :link '(url-link "https://github.com/SystemCrafters/rational-emacs")
   :group 'emacs)
 
+;;; package configuration
+(defun rational-package-archives-stale-p ()
+  "Check on-disk archives last modified date to see if it is stale.
+
+Return non-nil if the on-disk cache is older than one day or
+`nil' otherwise."
+  (let ((today (time-to-days (current-time)))
+        result)
+   (dolist (archive package-archives result)
+     (let* ((archive-name (expand-file-name (format "archives/%s/archive-contents" archive)
+                                            package-user-dir))
+            (archive-modified-date (time-to-days
+                                    (file-attribute-modification-time
+                                     (file-attributes archive-name)))))
+       (when (time-less-p archive-modified-date today)
+         (setq result t))))))
+
+(defmacro rational-install-package (package)
+  "Use `package.el' when it is enabled. Switch to using
+`straight.el' if it is not."
+  (if package-enable-at-startup
+      `(unless (package-installed-p ,package) (package-install ,package))
+    `(straight-use-package ,package)))
```

Comment:
While falling back to straight could be an interesting idea, straight will not work. The bootstrap code for straight is removed below. Is the expectation here that straight could still be used if the user decides to set it up on their own?

Reply:
I'd rather get rid of it. Initially I thought it would be good to have something like this, but I think I agree that it confuses things. I'll update the PR and remove it.